account_asset_management/models/account_asset.py

```diff
@@ -0,0 +1,1103 @@
+                asset.compute_depreciation_board()
+                # extra context to avoid recursion
+                ctx = dict(self._context, asset_validate_from_write=True)
+                asset.with_context(ctx).validate()
+        return True
+
+    def _create_first_asset_line(self):
+        self.ensure_one()
+        if self.depreciation_base and not self.depreciation_line_ids:
+            asset_line_obj = self.env['account.asset.line']
+            line_name = self._get_depreciation_entry_name(0)
+            asset_line_vals = {
+                'amount': self.depreciation_base,
+                'asset_id': self.id,
+                'name': line_name,
+                'line_date': self.date_start,
+                'init_entry': True,
+                'type': 'create',
+            }
+            asset_line = asset_line_obj.create(asset_line_vals)
+            if self.env.context.get('create_asset_from_move_line'):
+                asset_line.move_id = self._context['move_id']
+
+    @api.multi
+    def unlink(self):
+        for asset in self:
+            if asset.state != 'draft':
+                raise UserError(
+                    _("You can only delete assets in draft state."))
+            if asset.depreciation_line_ids.filtered(
+                    lambda r: r.type == 'depreciate' and r.move_check):
+                raise UserError(
+                    _("You cannot delete an asset that contains "
+                      "posted depreciation lines."))
+        # update accounting entries linked to lines of type 'create'
+        amls = self.with_context(
+            allow_asset_removal=True, from_parent_object=True
+        ).mapped('account_move_line_ids')
+        amls.write({'asset_id': False})
+        return super(AccountAsset, self).unlink()
+
+    @api.model
+    def name_search(self, name, args=None, operator='ilike', limit=100):
+        args = args or []
+        domain = []
+        if name:
+            domain = ['|',
+                      ('code', '=ilike', name + '%'),
+                      ('name', operator, name)]
+            if operator in expression.NEGATIVE_TERM_OPERATORS:
+                domain = ['&', '!'] + domain[1:]
+        assets = self.search(domain + args, limit=limit)
+        return assets.name_get()
+
+    @api.multi
+    @api.depends('name', 'code')
+    def name_get(self):
+        result = []
+        for asset in self:
+            name = asset.name
+            if asset.code:
+                name = ' - '.join([asset.code, name])
+            result.append((asset.id, name))
+        return result
+
+    @api.multi
+    def validate(self):
+        for asset in self:
+            if asset.type == 'normal' and asset.company_currency_id.is_zero(
+                    asset.value_residual):
+                asset.state = 'close'
+            else:
+                asset.state = 'open'
+        return True
+
+    @api.multi
```

Comment:
Well, if you want to call this method through an RPC call, I'm quite sure the `api.multi` decorator will be required.
`self.ensure_one` doesn't change anything as it will simply raise an error if it's called for multiple records.